wp/plugin/Makefile

```diff
@@ -1,22 +1,31 @@
-PASS_NAME = wp
+PASS_NAME := wp
+SHELL=/bin/bash
 
-SAMPLE_BIN_DIR = ../resources/sample_binaries
-SAVE_PROJECT_DIR = ./save_project
+SAMPLE_BIN_DIR := ../resources/sample_binaries
+SAVE_PROJECT_DIR := ./save_project
 API_PATH = $(shell bap --api-list-paths)
 
-.PHONY: all build install clean verifier save_project
+BUILD := _build/build_unique_id
 
-all: install
+.PHONY: all verifier save_project test
 
-build:
-	bapbuild -use-ocamlfind -pkgs 'z3,bap_wp,re' -tag 'warn(A-48-44)' $(PASS_NAME).plugin
+all: install
 
-install: build verifier save_project
+install: verifier save_project $(BUILD)
 	bapbundle update -desc 'Computes the weakest precondition of a subroutine given a postcondition.' $(PASS_NAME).plugin
 	bapbundle install $(PASS_NAME).plugin
 
+build: $(BUILD)
+
+$(BUILD): wp.ml
+	bapbuild -use-ocamlfind -pkgs 'z3,bap_wp,re' -tag 'warn(A-48-44)' $(PASS_NAME).plugin
+	# hack but no easy way around it
+	touch $(BUILD)
+
 test: install save_project dummy
-	ocamlbuild -r -use-ocamlfind -pkgs 'bap,z3,bap_wp,oUnit,re' -tag 'warn(A-48-44),debug,thread' -I tests test.native
+	ocamlbuild -r -use-ocamlfind \
+		-pkgs 'bap,z3,bap_wp,oUnit,re' \
+		-tag 'warn(A-48-44),debug,thread' -I tests test.native
 	./test.native
 
 verifier:
```

Comment:
Should we update the target for the verifier since we know where we are copying the files to?

Reply:
I'm also wondering if we should include removing the verifier API when we uninstall. Right now, even if we uninstall, that file is still located in the api path.